ixwebsocket/IXWebSocketTransport.cpp

```diff
@@ -176,33 +189,67 @@ namespace ix
         _onCloseCallback = onCloseCallback;
     }
 
-    // Only consider send time points for that computation.
-    // The receive time points is taken into account in Socket::poll (second parameter).
+    // Only consider send PING time points for that computation.
     bool WebSocketTransport::heartBeatPeriodExceeded()
     {
-        std::lock_guard<std::mutex> lock(_lastSendTimePointMutex);
+        std::lock_guard<std::mutex> lock(_lastSendPingTimePointMutex);
         auto now = std::chrono::steady_clock::now();
-        return now - _lastSendTimePoint > std::chrono::seconds(_heartBeatPeriod);
+        return now - _lastSendPingTimePoint > std::chrono::seconds(_heartBeatPeriod);
     }
 
-    void WebSocketTransport::poll()
+    bool WebSocketTransport::pongReceiveDelayExceeded()
     {
-        _socket->poll(
-            [this](PollResultType pollResult)
+        if (_heartBeatFactorDisconnectOnNoResponse == kDefaultHeartBeatFactorDisconnectOnNoResponse)
```

Comment:
If `kDefaultHeartBeatFactorDisconnectOnNoResponse` will be changed in future to 2 for example, this code will not work

Reply:
I think we won't change `kDefaultHeartBeatFactorDisconnectOnNoResponse`because it is -1 to disable by default

Reply:
currently yes but let's imagine in future we will enable it bu default